### PR TITLE
#166272294 Codeclimate test coverage

### PR DESCRIPTION
**Pull request by emmygozi**
#### What does this PR do?

This PR runs Codeclimate after passing the test from CircleCI
#### Description of Task to be completed?

- This PR run Codeclimate test coverage after passing the tests from Circleci
- This PR integrates Circleci with Slack.
- This PR also tests the app entry point ```App.js``` and install test dependencies.
#### How should this be manually tested?

`git clone https://github.com/andela/kingsmen-ah-frontend.git` 
`npm install`
`npm run test:coverage`
#### Any background context you want to provide?

N/A
#### What are the relevant pivotal tracker stories?

https://www.pivotaltracker.com/story/show/166272294
#### Screenshots (if appropriate)

#
image
### Questions:

N/A

### CODE REVIEW

**Review by despeauxz — CHANGES_REQUESTED** [review]
👍 , but  you've not set up where you specify the files it gets its coverage from and since we are using alias, it does not know how to resolve the path, might want to specify the same alias as for the `src` in the `moduleNameWrapper` in `jest.config.js`

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          command: npm run jest:coverage
+          environment:
+            JEST_JUNIT_OUTPUT: "reports/junit/js-test-results.xml"
+      - store_artifacts: # special step to save test results as as artifact
+          # Upload test summary for display in Artifacts: https://circleci.com/docs/2.0/artifacts/
+          path: reports/junit
+          prefix: tests
+      - store_test_results: # for display in Test Summary: https://circleci.com/docs/2.0/collect-test-data/
+          path: reports/junit
+      # See https://circleci.com/docs/2.0/deployment-integrations/ for deploy examples
```
Comment (GerrardE) [inline .circleci/config.yml]: give a space

File: junit.xml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<testsuites name="jest tests" tests="1" failures="0" time="2.04">
+  <testsuite name="SAMPLE TEST" errors="0" failures="0" skipped="0" timestamp="2019-07-04T00:01:55" time="0.825" tests="1">
+    <testcase classname="SAMPLE TEST should return a successful sample test " name="SAMPLE TEST should return a successful sample test " time="0.002">
+    </testcase>
+  </testsuite>
+</testsuites>
```
Comment (GerrardE) [inline junit.xml]: a line space here too

Comment (fidelisojeah) [inline junit.xml]: new line

File: .circleci/config.yml (hunk 1 above)
Comment (fidelisojeah) [inline .circleci/config.yml]: new line

File: package.json [hunk 3]
```diff
@@ -53,6 +56,9 @@
     "dotenv": "^8.0.0",
     "file-loader": "^4.0.0",
     "html-webpack-plugin": "^3.2.0",
+    "jest": "^24.8.0",
```
Comment (fidelisojeah) [inline package.json]: move to `devDependeicies`

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      - save_cache: # special step to save the dependency cache
+          key: dependency-cache-{{ checksum "package.json" }}
+          paths:
+            - ./node_modules
+      - run: # run tests
+          name: test
+          command: npm run test:coverage
+      - run:
+          name: Install JUnit coverage reporter
+          command: npm install jest-junit
```
Comment (fidelisojeah) [inline .circleci/config.yml]: This has already been installed, why install again?

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

File: junit.xml (hunk 2 above)
Comment (emmygozi) [reply junit.xml]: Fixing.

File: .circleci/config.yml (hunk 1 above)
Comment (emmygozi) [reply .circleci/config.yml]: image

Circleci/config.yml would throw an error if I create a newline as seen above.

Comment (emmygozi) [reply .circleci/config.yml]: image

Circleci/config.yml would throw an error if I create a new line as shown in the screenshot above.

Comment (fidelisojeah) [reply .circleci/config.yml]: Then you are doing something wrong...I would not merge this ticket with this error anyways

File: .gitignore [hunk 5]
```diff
@@ -97,3 +97,4 @@ fabric.properties
 .vscode/launch.json
 
 .env
+junit.xml
```
Comment (fidelisojeah) [inline .gitignore]: new line?

**Review by fidelisojeah — CHANGES_REQUESTED** [review]

**Pull request closed without merging**